Removes the old process and stats pages

matt-bernhardt opened the pull request:
** Why are these changes being introduced:

* Moving forward, this application will not need to provide the features
  provided by the /process and /stats pages (downloading files, marking
  a thesis as downloaded or withdrawn, providing a breakdown of stats
  for theses by these categories, etc). Additionally, the process_theses
  methods on the Thesis controller is going to be needed for the new
  thesis processing form.
* As a result, these elements should be removed from the application.
  This includes the process_theses and stats views, associated filtering
  partials, as well as the methods on the thesis controller. The thesis
  model itself will be untouched.

** Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETD-313

** How does this address that need:

* This removes the /process and /stats views, controller methods, and
  tests.

** Document any side effects to this change:

* The top link in the administrate UI now points to the home page, not
  the /process page.
* The logic determining where users land after logging in has been
  updated (app/controllers/application/controller.rb)
* Tests related to post-login redirects have been updated.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

YES (two gems removed)

JPrevost reviewed (changes requested):
I think there are two dependencies and possibly some css (less sure about that) we can remove along with this.

Take a look at this old PR for the two Gems it introduced and the css that may only have been used for this (although I'm not positive on that). (You'll need to remove the reference to those gems in application.js as well).

https://github.com/MITLibraries/thing/commit/e33ec29627a45e85ee6f4226b735737bd40d1c0f

matt-bernhardt commented:
Ok, I've removed those two gems, and the lines from `application.js` that loaded their javascripts for the stats page. The styles that were added as part of that commit don't appear to still be in the stylesheet, so there was nothing that I could see to remove. Thanks for catching this.

I'll update the PR text above to indicate that dependencies have now changed.
